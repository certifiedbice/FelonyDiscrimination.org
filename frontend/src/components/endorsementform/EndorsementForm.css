#organization-endorsement-form-container {
    text-align: center;
}

#endorsement-form legend {
    width: 100%;
}

#endorsement-form legend h3 {
    text-align: center;
}

.endorsement-form-element-container {
    margin: 0 0 10px 0;
}

.endorsement-form-element-container label {
    display: block;
    margin: 0 0 5px 20px;
    text-align: left;
    font-size: 12px;
}

.endorsement-form-element {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

.endorsement-form-element-container select {
    width: 95%;
    border-radius: 50px;
    padding: 5px;
}

#endorsement-form-submit {
    width: 90%;
    background-color: #4378bc;
    border: 1px solid #e6e7e8;
    border-radius: 50px;
    padding: 5px;
    font-size: 12px;
    color: #e6e7e8;
    cursor: pointer;
}

#endorsement-form-submit:hover {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

#endorsement-login-message, #endorsement-login-message h3 {
    width: 100%;
    text-align: center;
    color: #4378bc;
}

/* Desktop CSS */
@media(min-width:991px){
    
   .endorsement-form-element-container label {
        margin: 0 0 10px 10px;
    }

    .endorsement-form-element-container select {
        width: 100%;
    }

    #endorsement-form-submit {
        width: 25%;
    }

    #endorsement-form-submit:hover {
        width: 25%;
    }
    
}