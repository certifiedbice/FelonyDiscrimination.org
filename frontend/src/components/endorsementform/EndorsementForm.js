import React,{Component} from 'react';
import {Link} from 'react-router-dom';
import EndorsementsApiService from '../../services/endorsements-api-service';
import TokenService from '../../services/token-service';
import './EndorsementForm.css';

export default class EndorsementForm extends Component{
    
    state={error:null};
    
    renderEndorsementForm(){
        return(
            <form id='endorsement-form' name='endorsement-form' aria-label='Endorsement form' onSubmit={this.handleSubmit}>
                    <fieldset>
                        <legend><h3>Submit an Endorsement</h3></legend>
                        <div className='form-element-container'>
                            <label htmlFor='endorsement-form-select'>Choose your Endorsement:</label>
                            <select id='endorsement-form-select' name='endorsement' required aria-labelledby='endorsement-form-select'>
                                <option value='1'>Positive</option>
                                <option value='0'>Negative</option>
                            </select>
                        </div>
                        <div className='form-error-container'></div>
                        <div className='form-element-container'>
                            <input id='form-submit' type='submit' required aria-label='form-submit' value='Submit'/>
                        </div>
                    </fieldset>
                </form>
        );
    }
    
    renderEndorsementLoginMessage(){
        return(<div id='endorsement-login-message'><h3>You must <Link to='/login'>log in</Link> to submit endorsements</h3></div>);
    }
    
    handleSubmit=e=>{
        e.preventDefault()
        const {endorsement}=e.target;
        // Need to code a check against the endorsements table to look for existing user_id/org_id endorsement
        // return a notification that the user has already submitted an endorsement if exists
        EndorsementsApiService.postOrgEndorsements(this.props.orgId,endorsement.value)
            .then(res=>this.props.updateEndorsements(endorsement.value))
            .catch(this.state.error);
    }
    
    render(){
        return(
            <div id='form-container'>
                {TokenService.hasAuthToken()?this.renderEndorsementForm():this.renderEndorsementLoginMessage()}
            </div>
        );
	}

}