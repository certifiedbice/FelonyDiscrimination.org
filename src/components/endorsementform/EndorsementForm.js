import React,{Component} from 'react'
import EndorsementsApiService from '../../services/endorsements-api-service';
import './EndorsementForm.css';

export default class EndorsementForm extends Component{
    state={setError:null};
    handleSubmit=e=>{
        e.preventDefault()
        const {endorsement}=e.target;
        // Have to add in userId when user login system is implemented.
        // Need to code a check against the endorsements table to look for existing user_id/org_id endorsement
        // return a notification that the user has already submitted an endorsement if exists
        EndorsementsApiService.postOrgEndorsements(69,this.props.orgId,endorsement.value)
            .then(res=>this.props.updateEndorsements(endorsement.value))
            .catch(this.state.setError);
    }
    render(){
        return(
            <div id='organization-endorsement-form-container'>
                <form id="endorsement-form" name="endorsement-form" aria-label="Endorsement form" onSubmit={this.handleSubmit}>
                    <fieldset>
                        <legend><h3>Submit an Endorsement</h3></legend>
                        <div className="endorsement-form-element-container">
                            <label>Choose your Endorsement:</label>
                            <select className="endorsement-form-element" name="endorsement" id="endorsement-form-select" required>
                                <option value='1'>Positive</option>
                                <option value='0'>Negative</option>
                            </select>
                        </div>
                        <div className="endorsement-form-error-container"></div>
                        <div className="endorsement-form-element-container">
                            <input id="endorsement-form-submit" type="submit" required aria-label="endorsement form submit" value="Submit"/>
                        </div>
                    </fieldset>
                </form>
            </div>
        );
	}
}