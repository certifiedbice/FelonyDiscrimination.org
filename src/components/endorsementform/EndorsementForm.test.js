import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import EndorsementForm from './EndorsementForm';

let orgId='5';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<EndorsementForm orgId={orgId}/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<EndorsementForm orgId={orgId}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});