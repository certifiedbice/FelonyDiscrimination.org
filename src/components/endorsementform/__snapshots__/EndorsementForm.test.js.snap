// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="organization-endorsement-form-container"
>
  <form
    aria-label="Endorsement form"
    id="endorsement-form"
    name="endorsement-form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        <h3>
          Submit an Endorsement
        </h3>
      </legend>
      <div
        className="endorsement-form-element-container"
      >
        <label>
          Choose your Endorsement:
        </label>
        <select
          className="endorsement-form-element"
          id="endorsement-form-select"
          name="endorsement"
          required={true}
        >
          <option
            value="1"
          >
            Positive
          </option>
          <option
            value="0"
          >
            Negative
          </option>
        </select>
      </div>
      <div
        className="endorsement-form-error-container"
      />
      <div
        className="endorsement-form-element-container"
      >
        <input
          aria-label="endorsement form submit"
          id="endorsement-form-submit"
          required={true}
          type="submit"
          value="Submit"
        />
      </div>
    </fieldset>
  </form>
</div>
`;
